### DISCUSSION TITLE
Will this work with Vanilla Java?

### DISCUSSION THREAD

**Discussion by ValorZard** [wgpu4k/wgpu4k#36] [Q&A] [answered]
I've been interested in the recent improvements with Java with virtual threads and the upcoming project Valhalla, so I would like to use this library with vanilla Java if possible.

**Comment by ygdrasil-io** [comment, marked as answer]
Absolutely! Kotlin targeting the JVM is fully interoperable with Java. This means if you use the JVM variant of this library, your code stays within the JVM world. 

I might provide a full Java sample in a future preview, but currently, my focus is on iOS and Android support.

For your information, there is a separate Panama Glue version available here and used on the main kotlin library : 
https://mvnrepository.com/artifact/io.ygdrasil/wgpu4k-panama
https://github.com/wgpu4k/wgpu4k-panama

This part is quite raw and lacks features like library loading or a user-friendly API. I recommend sticking with the Kotlin library.
